### PR TITLE
dynamicsumary + examples + readme

### PR DESCRIPTION
**Pull request by artafres**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+      "description": "List all the summaries.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of summaries.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of summaries.",
```
Comment (rkmanda) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:243]: you dont need this to be under "value" . Just make it an array of Summary directly

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SummaryList": {
+      "description": "List all the summaries.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
```
Comment (rkmanda) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:237]: >nextLink [](http://example.com/codeflow?start=9&length=8)

nextLink is already on the wrapper

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+      "type": "object"
+    },
+    "SummaryProperties": {
+      "description": "Describes summary properties",
+      "properties": {
+        "summaryId": {
+          "description": "The id (a Guid) of the summary",
+          "type": "string"
+        },
+        "summaryName": {
```
Comment (rkmanda) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:277]: >summaryName [](http://example.com/codeflow?start=9&length=11)

top level envelope already has a name

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+        },
+        "isDeleted": {
+          "description": "A flag that indicates if the summary is deleted or not",
+          "type": "boolean"
+        },
+        "tenantId": {
+          "description": "The tenantId where the summary belongs to",
+          "type": "string"
+        },
+        "summaryStatus": {
```
Comment (rkmanda) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:332]: >summaryStatus [](http://example.com/codeflow?start=9&length=13)

make this an enum

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+          "description": "The status of the Summary : Active, Deleted",
+          "type": "string"
+        },
+        "uploadStatus": {
```
Comment (rkmanda) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:336]: >uploadStatus [](http://example.com/codeflow?start=9&length=12)

make this an enum

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+        },
+        "summaryDescription": {
+          "description": "The description for the summary",
+          "type": "string"
+        },
+        "rawContent": {
+          "description": "The raw content that represents to summary items to create.",
+          "type": "string"
+        },
+        "type": {
```
Comment (rkmanda) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:320]: >type [](http://example.com/codeflow?start=9&length=4)

enum?

Comment (artafres) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:320]: not an Enum. it's a constant string value to indicate the type of the document

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json (hunk 2 above)
Comment (artafres) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:237]: removing it results in error when running autorest validation. the x-ms-pageable requires to have nextLink defined. looking at other *.json (other sendpoints) in the same directory, they all defined the nextLink.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json (hunk 3 above)
Comment (artafres) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:277]: not sure if I understood the issue here. the DynamicSummary resource has a property "summaryName" which user defines that (and it is editable) to name the summary. the summary has summaryId which is in GUID format and is uniqe and not editable

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json (hunk 5 above)
Comment (artafres) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json]: modified to provisioningState

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json (hunk 1 above)
Comment (artafres) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:243]: you dont need **this** to be under "value". 
by "this" you mean the "items" field?

Comment (mentat9) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:243]: Looks correct to me. Ref: [https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html?q=nextLink#get-resource-collection](https://armwiki.azurewebsites.net/api_contracts/RPCstrongvalidation.html?q=nextLink#get-resource-collection)

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "summaryStatus"
+          }
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState",
```
Comment (mentat9) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:346]: >definitions/ProvisioningState [](http://example.com/codeflow?start=21&length=29)

Need to define this.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json (hunk 3 above)
Comment (mentat9) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json:277]: +1. This an ARM resource, which means the top level properties already include id and name (defined in the ResourceWithEtag $ref above). You shouldn't duplicate those properties within the property bag. If summaryId is different from the top-level "id" property and/or summaryName is different from the top-level "name" property, then these should get different names to make it clear what's different about them.

**Pull request closed without merging**